### PR TITLE
gh-132: Prepare routes for game switching

### PR DESCRIPTION
**Pull request by littlewhywhat**
After #138, before #135 

Only preparing the routes no game switching yet.

part of gh-132

### CODE REVIEW

File: frontend/src/Root.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  <Provider store={store}>
+    <Router>
+      <Switch>
+        <Route path='/foosball' component={App}/>
+        <Route>
+          <Redirect to='/foosball'/>
+        </Route>
+      </Switch>
+    </Router>
+  </Provider>
```
[the remark this answers is not in the document]
Comment (littlewhywhat) [reply frontend/src/Root.js]: I am Root! :grinning: I can agree with that though I dont mind either way :) let's say wait for others opinion

File: frontend/src/app/pages/CreateMatch/index.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
-import React, { Component } from 'react'
+import React from 'react'
 import { CreateMatch } from '../../components/CreateMatch/CreateMatch'
 
-export class CreateMatchPage extends Component {
-  render = () => <CreateMatch maxPlayerNumber={2} minPlayerNumber={1} />
-}
+export const CreateMatchPage = ({ constructUrl }) =>
+  <CreateMatch constructUrl={constructUrl} maxPlayerNumber={2} minPlayerNumber={1} />
```
[the remark this answers is not in the document]
Comment (littlewhywhat) [reply frontend/src/app/pages/CreateMatch/index.js]: I think it's not part of this issue - as you can see it was like that before.

File: frontend/src/app/pages/Dashboard/index.js [hunk 3]
```diff
@@ -10,26 +10,27 @@ import { BattleHistory } from './../../components/BattleHistory/BattleHistory'
...
-      </Box>
-    )
-  }
-}
 
-const mapStateToProps = state => ({
+const mapStateToProps = (state, { constructUrl }) => ({
   createMatchStatus: state.matchesStatus,
+  lastMatches: getLastMatches(state).map(match => createMatchWithPlayerLinks(match, constructUrl)),
+  topPlayers: getTopPlayers(state).map(player => createPlayerWithLink(player, constructUrl)),
```
[the remark this answers is not in the document]
Comment (littlewhywhat) [reply frontend/src/app/pages/Dashboard/index.js]: yes, I first made them as selector but then I realized that selectors are functions on state - and these ones are not selectors as they need another parameter - so I put those in utils. Not sure if I got your next sentence but if you mean that the way it is now we can't get last matches with links somewhere else without duplication ... I can agree with that :) what about making a func: `withPlayerLinks(matches)` and `withLinks(players)`?

File: frontend/src/app/pages/CreateMatch/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Fezzzi) [reply frontend/src/app/pages/CreateMatch/index.js]: I agree that such edit will probably fit more into some refactoring issue but as I didn't meet any such issue on the project yet, I decided to mention it anyway. Might be probably a good idea to create one to store these things and then resolve them all at once when enough is accumulated?

File: frontend/src/app/pages/Dashboard/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Fezzzi) [reply frontend/src/app/pages/Dashboard/index.js]: Wouldn't storing the route in the state resolve this problem? That way all infomation needed will be accessible from selectors. Or is there any problem with this approach that I can't see?

Yes, by my second sentence I tried to say something like that... :grinning:

File: frontend/src/app/pages/Leaderboard/index.js [hunk 4]
```diff
@@ -2,9 +2,18 @@ import React from 'react'
```
[the remark this answers is not in the document]
Comment (littlewhywhat) [reply frontend/src/app/pages/Leaderboard/index.js]: fixed in https://github.com/salsita/foosball-rating/pull/139/commits/f5ddc54aae5c4f0cfc27d9edae8578ceed628468

File: frontend/src/app/pages/MatchList/index.js [hunk 5]
```diff
@@ -1,11 +1,20 @@
 import React from 'react'
 import { BattleHistory } from '../../components/BattleHistory/BattleHistory'
 import { Subtitle, Box } from './../../../styles/blocks'
+import { connect } from 'react-redux'
+import { getLastMatches } from '../../modules/matches/matches-selectors'
+import { createMatchWithPlayerLinks } from '../../modules/matches/matches-utils'
 
-export const MatchListPage = () =>
+export const MatchListPageComponent = ({ lastMatches }) =>
```
[the remark this answers is not in the document]
Comment (littlewhywhat) [reply frontend/src/app/pages/MatchList/index.js]: fixed in https://github.com/salsita/foosball-rating/pull/139/commits/f5ddc54aae5c4f0cfc27d9edae8578ceed628468

File: frontend/src/app/pages/Leaderboard/index.js [hunk 6]
```diff
@@ -2,9 +2,18 @@ import React from 'react'
 
 import { Leaderboard } from '../../components/Leaderboard/Leaderboard'
 import { Subtitle, Box } from './../../../styles/blocks'
+import { connect } from 'react-redux'
+import { getTopPlayers } from '../../modules/players/players-selectors'
+import { createPlayerWithLink } from '../../modules/players/players-utils'
 
-export const LeaderboardPage = () =>
+export const LeaderboardPageComponent = ({ topPlayers }) =>
```
[the remark this answers is not in the document]
Comment (littlewhywhat) [reply frontend/src/app/pages/Leaderboard/index.js]: fixed in https://github.com/salsita/foosball-rating/pull/139/commits/f5ddc54aae5c4f0cfc27d9edae8578ceed628468

File: frontend/src/app/pages/CreateMatch/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (littlewhywhat) [reply frontend/src/app/pages/CreateMatch/index.js]: fixed in https://github.com/salsita/foosball-rating/pull/139/commits/85f201dfd0598532d17c6687675469d65638fcdd

File: frontend/src/app/pages/Dashboard/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (littlewhywhat) [reply frontend/src/app/pages/Dashboard/index.js]: it's not a recommended practice https://reacttraining.com/react-router/core/guides/redux-integration/deep-integration. If needed though I would have to use yet another framework. In general it's not that simple to synchronize store and Router.

[the remark this answers is not in the document]
Comment (littlewhywhat) [reply frontend/src/app/pages/Dashboard/index.js]: add withLinks, withPlayerLinks https://github.com/salsita/foosball-rating/pull/139/commits/edb607759dde3785619c3397e5754338b1b3b951

[the remark this answers is not in the document]
Comment (Fezzzi) [reply frontend/src/app/pages/Dashboard/index.js]: > If needed though I would have to use yet another framework. In general it's not that simple to synchronize store and Router.

I see, in that case, `withLinks` and `withPlayerLinks` is more than sufficient. Thanks!

File: frontend/src/app/pages/CreateMatch/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Fezzzi) [reply frontend/src/app/pages/CreateMatch/index.js]: Great, thanks!

File: frontend/src/Root.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (littlewhywhat) [reply frontend/src/Root.js]: @novellizator @petrk-salsita need your opinion here, thanks!

[the remark this answers is not in the document]
Comment (novellizator) [reply frontend/src/Root.js]: this is super low priority. My personal preference is to use App, but fine both ways

[the remark this answers is not in the document]
Comment (littlewhywhat) [reply frontend/src/Root.js]: then let's leave it as it is

**Pull request merged**